src/main/java/it/gov/pagopa/ranking/service/initiative/ranking/retrieve/OnboardingRankingRuleEndedRetrieverServiceImpl.java

```diff
@@ -0,0 +1,76 @@
+package it.gov.pagopa.ranking.service.initiative.ranking.retrieve;
+
+import com.azure.messaging.servicebus.ServiceBusClientBuilder;
+import com.azure.messaging.servicebus.ServiceBusReceivedMessage;
+import com.azure.messaging.servicebus.ServiceBusReceiverClient;
+import com.azure.messaging.servicebus.administration.ServiceBusAdministrationClient;
+import com.azure.messaging.servicebus.administration.models.QueueRuntimeProperties;
+import it.gov.pagopa.ranking.dto.initiative.filter.OnboardingRequestPendingDTO;
+import it.gov.pagopa.ranking.model.InitiativeConfig;
+import it.gov.pagopa.ranking.model.RankingStatus;
+import it.gov.pagopa.ranking.service.initiative.InitiativeConfigService;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDate;
+import java.util.List;
+
+@Service
+@Slf4j
+public class OnboardingRankingRuleEndedRetrieverServiceImpl implements OnboardingRankingRuleEndedRetrieverService {
+    private final InitiativeConfigService initiativeConfigService;
+    private final long beforeDays;
+    private final String connectionString;
+    private final String queueName;
+    private final ServiceBusAdministrationClient adminClient;
+
+    public OnboardingRankingRuleEndedRetrieverServiceImpl(InitiativeConfigService initiativeConfigService,
+                                                          @Value("${app.ranking-build-file.retrieve-initiative.day-before}") long beforeDays,
+                                                          @Value("${app.service-bus.onboarding-request-pending.string-connection}") String connectionString,
+                                                          @Value("${app.service-bus.onboarding-request-pending.destination}") String queueName,
+                                                          ServiceBusAdministrationClient adminClient) {
+        this.initiativeConfigService = initiativeConfigService;
+        this.beforeDays = beforeDays;
+        this.connectionString = connectionString;
+        this.queueName = queueName;
+        this.adminClient = adminClient;
+    }
+
+    @Override
+    public List<InitiativeConfig> retrieve(){
+        LocalDate now = LocalDate.now();
+        return initiativeConfigService.findByRankingStatusRankingEndDateBetween(RankingStatus.WAITING_END, now.minusDays(beforeDays+1), now)
```

Comment:
The Between operation includes the extremes of the range configured. we are interested into
now-dayBefore <= endDate < now
```suggestion
        return initiativeConfigService.findByRankingStatusRankingEndDateBetween(RankingStatus.WAITING_END, now.minusDays(beforeDays), now.minusDays(1))
```

Reply:
In mongo repository the between is exclusive
https://docs.spring.io/spring-data/mongodb/docs/1.2.0.RELEASE/reference/html/mongo.repositories.html